app/src/main/java/org/piramalswasthya/cho/repositories/OphthalmicRepository.kt

```
@@ -27,4 +54,113 @@ class OphthalmicRepository @Inject constructor(
+                            }
+                        } else if (responseStatusCode == 5002) {
+                            if (userRepo.refreshTokenTmc(user.userName, user.password)) {
+                                throw SocketTimeoutException()
+                            }
+                        } else if (responseStatusCode == 5000) {
+                            return@withContext true
+                        }
+                    }
+                }
+                return@withContext false
+            } catch (e: SocketTimeoutException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)